### ISSUE TITLE
[Model Unload]If model Unloaded, still trace setting api works 

### ISSUE THREAD

**Issue by Kanupriyagoyal** [triton-inference-server/server#6154]
**Description**
When a model is unloaded via triton's unload API, it is observed that the other functions / APIs related to same model still works creating an improper API workflow.
```
1 http_server.cc:3460] HTTP request: 2 /v2/repository/models/model_1/unload
1 model_lifecycle.cc:579] successfully unloaded 'model_1' version 0
1 http_server.cc:3460] HTTP request: 2 /v2/models/model_1/trace/setting
{
    "trace_level": [
        "OFF"
    ],
    "trace_rate": "1",
    "trace_count": "-1",
    "log_frequency": "0",
    "trace_file": ""
}

/config
{
    "error": "Request for unknown model: model_1 has no available versions"
}

```
**Triton Information**
What version of Triton are you using?
r23.04

Are you using the Triton container or did you build it yourself?
Same instructions using build.py

**To Reproduce**
1. Invoke POST /v2/repository/models/model_1/unload API.
2. Invoke GET /v2/models/model_1/config API. Observe the response, it is clear that the model is unloaded and hence, the config API isn’t working for model_1.
As model_1 model is unloaded in step 1.
3. Invoke POST /v2/models/model_1/trace/setting API with "log_frequency" value 1 and observe that "log_frequency" value is updated to 1. Server is not validating whether model_1 model is available or not.

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

**Expected behavior**
The Server should ideally respond with "400" or any generic error similar behaviour for model api.

**Comment by rmccorm4** [comment]
The reported issue here is that users can still access/change trace settings for a previously existing model that was unloaded. I don't know what our stance is on this case yet, as we do keep around certain model/version metadata even for models that are unloaded for the sake of reporting in the `/repository/index` APIs. 

I'd have to take a look at how we're mapping the trace settings and we can decide the expected behavior for unloaded models. CC @kthui @GuanLuo I think this is loosely related to #5944 in how we keep models/version metadata around after unloads.

---

However, it looks like there's some weirder behavior in the trace API for model-specific trace settings. It looks like we're not validating if the model exists or not. I can both GET and POST the trace settings for a non-existent model that was never loaded nor in the model repository before:
```bash
curl -s localhost:8000/v2/models/does-not-exist/trace/setting | jq
curl -s -X POST localhost:8000/v2/models/does-not-exist/trace/setting -d '{"log_frequency": "1"}' | jq
curl -s localhost:8000/v2/models/does-not-exist/trace/setting | jq
```
CC @oandreeva-nv

**Comment by kpg993** [comment]
Hi @oandreeva-nv, @rmccorm4 .Will it be fixed in the upcoming release or consider it as design limitation. 
Thanks in advance!!

**Comment by rmccorm4** [comment]
Hi @kpg993 @Kanupriyagoyal, I believe this was fixed in the 23.09 release here: https://github.com/triton-inference-server/triton_cli/pull/27#event-11540899596

Here's an example running on the latest 23.12 release:
```bash
rmccormick@ced35d0-lcedt:~$ curl -s localhost:8000/v2/models/does-not-exist/trace/setting | jq
{
  "error": "Request for unknown model : does-not-exist"
}
rmccormick@ced35d0-lcedt:~$ curl -s -X POST localhost:8000/v2/models/does-not-exist/trace/setting -d '{"log_frequency": "1"}' | jq
{
  "error": "Request for unknown model : does-not-exist"
}
```

**Comment by rmccorm4** [comment]
I'm going to close this per the previous comment, but please let us know or open a new issue (referencing this one) if you run into any issues with it.
